sdk/helper/awsutil/generate_credentials.go

```diff
@@ -51,6 +55,8 @@ func (c *CredentialsConfig) GenerateCredentialChain() (*credentials.Credentials,
 				SecretAccessKey: c.SecretKey,
 				SessionToken:    c.SessionToken,
 			}})
+		c.Logger.Debug("added static credential provider", "AccessKey", c.AccessKey)
```

Comment:
`c.Logger` could potentially be nil if none is passed into `CredentialsConfig` so these calls would panic. Let's wrap logger calls with a nil check.